tests/cross/src/test/scala/tests/pc/InlayHintsSuite.scala

```diff
@@ -16,6 +16,302 @@ class InlayHintsSuite extends BaseInlayHintsSuite {
     }
   }
 
+  check(
+    "single-chain-same-line",
+    """object Main{
+      |  trait Bar {
+      |   def bar: Bar
+      |  }
+      |
+      |  trait Foo {
+      |    def foo(): Foo
+      |  }
+      |
+      |val bar: Bar = ???
+      |val foo: Foo = ???
+      |
+      |val thing1: Bar = bar.bar
+      |val thing2: Foo = foo.foo()
+      |}
+      |""".stripMargin,
+    """object Main{
+      |  trait Bar {
+      |   def bar: Bar
+      |  }
+      |
+      |  trait Foo {
+      |    def foo(): Foo
+      |  }
+      |
+      |val bar: Bar = ???
+      |val foo: Foo = ???
+      |
+      |val thing1: Bar = bar.bar
+      |val thing2: Foo = foo.foo()
+      |}
+      |""".stripMargin
+  )
+
+  check(
+    "multi-chain-same-line",
+    """object Main{
+      |  trait Bar {
+      |   def bar: Bar
+      |  }
+      |
+      |  trait Foo {
+      |    def foo(): Foo
+      |  }
+      |
+      |val bar: Bar = ???
+      |val foo: Foo = ???
+      |
+      |val thing1: Bar = bar.bar.bar
+      |val thing2: Foo = foo.foo().foo()
+      |}
+      |""".stripMargin,
+    """object Main{
+      |  trait Bar {
+      |   def bar: Bar
+      |  }
+      |
+      |  trait Foo {
+      |    def foo(): Foo
+      |  }
+      |
+      |val bar: Bar = ???
+      |val foo: Foo = ???
+      |
+      |val thing1: Bar = bar.bar.bar
+      |val thing2: Foo = foo.foo().foo()
+      |}
+      |""".stripMargin
+  )
+
+  check(
+    "single-chain-new-line",
+    """object Main{
+      |  trait Bar {
+      |   def bar: Bar
+      |  }
+      |
+      |  trait Foo {
+      |    def foo(): Foo
+      |  }
+      |
+      |val bar: Bar = ???
+      |val foo: Foo = ???
+      |
+      |val thing1: Bar = bar
+      |  .bar
+      |val thing2: Foo = foo
+      |  .foo()
+      |}
+      |""".stripMargin,
+    """object Main{
+      |  trait Bar {
+      |   def bar: Bar
+      |  }
+      |
+      |  trait Foo {
+      |    def foo(): Foo
+      |  }
+      |
+      |val bar: Bar = ???
+      |val foo: Foo = ???
+      |
+      |val thing1: Bar = bar
+      |  .bar/*: Bar<<(2:8)>>*/
+      |val thing2: Foo = foo
+      |  .foo()/*: Foo<<(6:8)>>*/
+      |}
+      |""".stripMargin
+  )
+
+  check(
+    "simple-chain",
+    """object Main{
+      |  trait Foo {
+      |   def bar: Bar
+      |  }
+      |
+      |  trait Bar {
+      |    def foo(): Foo
+      |  }
+      |
+      |val foo: Foo = ???
+      |
+      |val thingy: Bar = foo
+      |  .bar
+      |  .foo()
+      |  .bar
+      |}
+      |""".stripMargin,
+    """object Main{
+      |  trait Foo {
+      |   def bar: Bar
+      |  }
+      |
+      |  trait Bar {
+      |    def foo(): Foo
+      |  }
+      |
+      |val foo: Foo = ???
+      |
+      |val thingy: Bar = foo
+      |  .bar/*  : Bar<<(6:8)>>*/
+      |  .foo()/*: Foo<<(2:8)>>*/
+      |  .bar/*  : Bar<<(6:8)>>*/
```

Comment:
getting all of the inlay hints for a block justified alongside each other feels so good. Please don't make me get rid of it orz